clang/test/CIR/CodeGen/three-way-comparison-noncanonical.cpp

```diff
@@ -0,0 +1,42 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -std=c++20 -fclangir-enable -emit-cir -mmlir --mlir-print-ir-before=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=BEFORE
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -std=c++20 -fclangir-enable -emit-cir -mmlir --mlir-print-ir-after=cir-lowering-prepare %s -o %t.cir 2>&1 | FileCheck %s -check-prefix=AFTER
+
+#include "Inputs/std-compare-noncanonical.h"
+
+auto three_way_strong(int x, int y) {
+  return x <=> y;
+}
```

Comment:
Can this test live along side others in `clang/test/CIR/CodeGen/three-way-comparison.cpp` or there are additional complications?

Reply:
Same as above. The key difference between these two tests is the value `cir.cmp3way` produced for lt, eq, and gt. These result values are defined by `std-compare.h` and `std-compare-noncanonical.h`. Unsure about how to merge them into one single definition.

Reply:
Updated.